### ISSUE TITLE
Difficulties with Early Bound Generator

### ISSUE THREAD

**Issue by PeulenF** [daryllabar/DLaB.Xrm.XrmToolBoxTools#43]
I was trying to create the early bound CRM entitties, using XrmToolBox v1.2016.4.28.

First I set up my connection to our CRM instance and after I opened the Early Bound Generator.
I setup my preferred options and the list of entities which to exclude. 
But once I try to create the entities, I get a list of options and the following error:

Output file was not updated or not found! + path to my Service context file.

My output file is located in the right directory. 

Any help into the solution of this problem is greatly appreciated.

**Comment by daryllabar** [comment]
Can I get the full output in the log?

**Comment by PeulenF** [comment]
CRM Authentication Type Detected: ActiveDirectory
Executing "..\XrmToolBox\Plugins\CrmSvcUtil Ref\crmsvcutil.exe" /namespace:"CrmLib" /out:"..\XrmToolBox\Plugins\CrmSvcUtil Ref\CrmServiceContext.cs" /servicecontextname:"CrmServiceContext" /codecustomization:"DLaB.CrmSvcUtilExtensions.Entity.CustomizeCodeDomService,DLaB.CrmSvcUtilExtensions" /codegenerationservice:"DLaB.CrmSvcUtilExtensions.Entity.CustomCodeGenerationService,DLaB.CrmSvcUtilExtensions" /codewriterfilter:"DLaB.CrmSvcUtilExtensions.Entity.CodeWriterFilterService,DLaB.CrmSvcUtilExtensions" /namingservice:"DLaB.CrmSvcUtilExtensions.NamingService,DLaB.CrmSvcUtilExtensions" /metadataproviderservice:"DLaB.CrmSvcUtilExtensions.Entity.MetadataProviderService,DLaB.CrmSvcUtilExtensions" 
CrmSvcUtil : CRM Service Utility [Version 8.0.1.7297]
¸ 2015 Microsoft Corporation. All rights reserved


Options:
 /nologo
  Suppresses the banner.

 /language:<language>
  The language to use for the generated proxy code.  This can be either 'CS' or 'VB'.  The default language is 'CS'.  Short form is '/l:'.

 /url:<url>
  A url or path to the SDK endpoint to contact for metadata.

 /out:<filename>
  The filename for the generated proxy code.  Short form is '/o:'.

 /namespace:<namespace>
  The namespace for the generated proxy code.  The default namespace is the global namespace.  Short form is '/n:'.

 /interactivelogin:
  Presents a login dialog to loginto the service with, if passed all other connect info is ignored.  Short form is '/il:'.

 /connectionstring:
  Connection String to use when connecting to CRM. If provided, all other connect info is ignored.  Short form is '/connstr:'.

 /username:<username>
  Username to use when connecting to the server for authentication.  Short form is '/u:'.

 /password:<password>
  Password to use when connecting to the server for authentication.  Short form is '/p:'.

 /domain:<domain>
  Domain to authenticate against when connecting to the server.  Short form is '/d:'.

 /serviceContextName:<service context name>
  The name for the generated service context. If a value is passed in, it will be used for the Service Context.  If not, no Service Context will be generated

 /help
  Show this usage message.  Short form is '/?'.

 /generateActions
  Generate wrapper classes for custom actions  Short form is '/a'.

 /deviceid:<deviceid>
  Device ID to use when connecting to the online server for authentication.  Short form is '/di:'.

 /devicepassword:<devicepassword>
  Device Password to use when connecting to the online server for authentication.  Short form is '/dp:'.


Example:
crmsvcutil.exe /url:http://localhost/Organization1/XRMServices/2011/Organization.svc /out:GeneratedCode.cs

Output file was not updated or not found!  ..\XrmToolBox\Plugins\CrmSvcUtil Ref\CrmServiceContext.cs

**Comment by daryllabar** [comment]
hmm... you're missing a URL parameter... not sure how that could have happened.

Has it worked for you before, or is the first time you've tried it?

**Comment by PeulenF** [comment]
It has worked on an older version. This is the first time I try it on this version.

**Comment by daryllabar** [comment]
What version of the Early Bound Generator is it?

**Comment by PeulenF** [comment]
Version: 1.2016.4.27

**Comment by daryllabar** [comment]
Can you delete (make a backup first if you wish) all DLaB files in your plugins directory, and the entire CrmSvcUtil Folder, and redownload?

**Comment by PeulenF** [comment]
I tried this but I get the same result as before.

**Comment by daryllabar** [comment]
Think I figured it out.  You're connecting with XRM Toolbox without entering credentials, just using AD correct?

**Comment by PeulenF** [comment]
Yes indeed.

**Comment by daryllabar** [comment]
[DLaB.Xrm.2016.zip](https://github.com/daryllabar/DLaB.Xrm.XrmToolBoxTools/files/274847/DLaB.Xrm.2016.zip)
Try downloading, unblocking, and extracting these files into your plugins directory of the XrmToolBox.  It should prompt you for a password, then you can enter it and everything should get populated.

**Comment by PeulenF** [comment]
I tried this but it causes the error shown below. Also trying to close the program or trying to specify which entities to skip, causes crashes. 
Could this be caused cause we are using CRM 2011?

CRM Authentication Type Detected: ActiveDirectory
Executing "..\XrmToolBox\Plugins\CrmSvcUtil Ref\crmsvcutil.exe" /namespace:"CrmLib" /out:"..\XrmToolBox\Plugins\CrmSvcUtil Ref\CrmServiceContext.cs" /servicecontextname:"CrmServiceContext" /codecustomization:"DLaB.CrmSvcUtilExtensions.Entity.CustomizeCodeDomService,DLaB.CrmSvcUtilExtensions" /codegenerationservice:"DLaB.CrmSvcUtilExtensions.Entity.CustomCodeGenerationService,DLaB.CrmSvcUtilExtensions" /codewriterfilter:"DLaB.CrmSvcUtilExtensions.Entity.CodeWriterFilterService,DLaB.CrmSvcUtilExtensions" /namingservice:"DLaB.CrmSvcUtilExtensions.NamingService,DLaB.CrmSvcUtilExtensions" /metadataproviderservice:"DLaB.CrmSvcUtilExtensions.Entity.MetadataProviderService,DLaB.CrmSvcUtilExtensions" /connectionstring:"Url=*CRMPath*;UserName=;Password=*************" 
CrmSvcUtil : CRM Service Utility [Version 8.0.1.7297]
¸ 2015 Microsoft Corporation. All rights reserved

Microsoft.Xrm.Tooling.Connector.CrmServiceClient Error: 2 : Source	: System.ServiceModel
Method	: Retrieve
Date	: 23/05/2016
Time	: 15:53:39
Error	: Metadata contains a reference that cannot be resolved: '../XRMServices/2011/Discovery.svc?wsdl&sdkversion=0'.
Stack Trace	: at System.ServiceModel.Description.MetadataExchangeClient.MetadataRetriever.Retrieve(TimeoutHelper timeoutHelper)
   at System.ServiceModel.Description.MetadataExchangeClient.ResolveNext(ResolveCallState resolveCallState)
   at System.ServiceModel.Description.MetadataExchangeClient.GetMetadata(MetadataRetriever retriever)
   at Microsoft.Xrm.Sdk.Client.ServiceMetadataUtility.RetrieveServiceEndpointMetadata(Type contractType, Uri serviceUri, Boolean checkForSecondary)
   at Microsoft.Xrm.Sdk.Client.ServiceConfiguration`1..ctor(Uri serviceUri, Boolean checkForSecondary)
   at Microsoft.Xrm.Sdk.Client.ServiceConfigurationFactory.CreateManagement[TService](Uri serviceUri, Boolean enableProxyTypes, Assembly assembly)
   at Microsoft.Xrm.Tooling.Connector.CrmWebSvc.CreateAndAuthenticateProxy[T](IServiceManagement`1 servicecfg, Uri ServiceUri, Uri homeRealm, ClientCredentials userCredentials, ClientCredentials deviceCredentials, String LogString)
   at Microsoft.Xrm.Tooling.Connector.CrmWebSvc.DiscoverOrganizations(Uri discoveryServiceUri, Uri homeRealmUri, ClientCredentials clientCredentials, ClientCredentials deviceCredentials)
   at Microsoft.Xrm.Tooling.Connector.CrmWebSvc.InitCRM2011Service()
======================================================================================================================
Inner Exception Level 1	: 
Source	: System
Method	: GetResponse
Date	: 23/05/2016
Time	: 15:53:39
Error	: The remote name could not be resolved: '..'
Stack Trace	: at System.Net.HttpWebRequest.GetResponse()
   at System.ServiceModel.Description.MetadataExchangeClient.MetadataLocationRetriever.DownloadMetadata(TimeoutHelper timeoutHelper)
   at System.ServiceModel.Description.MetadataExchangeClient.MetadataRetriever.Retrieve(TimeoutHelper timeoutHelper)
======================================================================================================================

Microsoft.Xrm.Tooling.Connector.CrmServiceClient Error: 2 : Unable to Login to Dynamics CRM
Unable to Login to Dynamics CRM
CrmSvcUtil Error: 2 : Exiting program with exit code 2 due to exception : System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Exception: Connection to CRM is not established. Aborting process.
   at Microsoft.Crm.Services.Utility.SdkMetadataProviderService.Microsoft.Crm.Services.Utility.IMetadataProviderService.LoadMetadata()
   at DLaB.CrmSvcUtilExtensions.Entity.MetadataProviderService..ctor(IMetadataProviderService defaultService, IDictionary`2 paramters)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Crm.Services.Utility.ServiceFactory.CreateInstance[TIService](TIService defaultServiceInstance, String parameterValue, CrmSvcUtilParameters parameters)
   at Microsoft.Crm.Services.Utility.ServiceProvider.InitializeServices(CrmSvcUtilParameters parameters)
   at Microsoft.Crm.Services.Utility.CrmSvcUtil.get_ServiceProvider()
   at Microsoft.Crm.Services.Utility.CrmSvcUtil.Run()
   at Microsoft.Crm.Services.Utility.CrmSvcUtil.Main(String[] args)
CrmSvcUtil Error: 2 : ===== DETAIL ======
CrmSvcUtil Error: 2 : Source	: mscorlib
Method	: InvokeMethod
Date	: 15:53:39
Time	: 23/05/2016
Error	: Exception has been thrown by the target of an invocation.
Stack Trace	: at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Crm.Services.Utility.ServiceFactory.CreateInstance[TIService](TIService defaultServiceInstance, String parameterValue, CrmSvcUtilParameters parameters)
   at Microsoft.Crm.Services.Utility.ServiceProvider.InitializeServices(CrmSvcUtilParameters parameters)
   at Microsoft.Crm.Services.Utility.CrmSvcUtil.get_ServiceProvider()
   at Microsoft.Crm.Services.Utility.CrmSvcUtil.Run()
   at Microsoft.Crm.Services.Utility.CrmSvcUtil.Main(String[] args)
======================================================================================================================
Inner Exception Level 1	: 
Source	: CrmSvcUtil
Method	: Microsoft.Crm.Services.Utility.IMetadataProviderService.LoadMetadata
Date	: 15:53:39
Time	: 23/05/2016
Error	: Connection to CRM is not established. Aborting process.
Stack Trace	: at Microsoft.Crm.Services.Utility.SdkMetadataProviderService.Microsoft.Crm.Services.Utility.IMetadataProviderService.LoadMetadata()
   at DLaB.CrmSvcUtilExtensions.Entity.MetadataProviderService..ctor(IMetadataProviderService defaultService, IDictionary`2 paramters)
======================================================================================================================

Output file was not updated or not found!  ..\XrmToolBox\Plugins\CrmSvcUtil Ref\CrmServiceContext.cs
CRM Authentication Type Detected: ActiveDirectory
Executing "..\XrmToolBox\Plugins\CrmSvcUtil Ref\crmsvcutil.exe" /namespace:"CrmLib" /out:"..\XrmToolBox\Plugins\CrmSvcUtil Ref\CrmServiceContext.cs" /servicecontextname:"CrmServiceContext" /codecustomization:"DLaB.CrmSvcUtilExtensions.Entity.CustomizeCodeDomService,DLaB.CrmSvcUtilExtensions" /codegenerationservice:"DLaB.CrmSvcUtilExtensions.Entity.CustomCodeGenerationService,DLaB.CrmSvcUtilExtensions" /codewriterfilter:"DLaB.CrmSvcUtilExtensions.Entity.CodeWriterFilterService,DLaB.CrmSvcUtilExtensions" /namingservice:"DLaB.CrmSvcUtilExtensions.NamingService,DLaB.CrmSvcUtilExtensions" /metadataproviderservice:"DLaB.CrmSvcUtilExtensions.Entity.MetadataProviderService,DLaB.CrmSvcUtilExtensions" /connectionstring:"Url=*CRMPath*;Domain=****;UserName=*********;Password=************" 
CrmSvcUtil : CRM Service Utility [Version 8.0.1.7297]
¸ 2015 Microsoft Corporation. All rights reserved

Microsoft.Xrm.Tooling.Connector.CrmServiceClient Error: 2 : Source	: System.ServiceModel
Method	: Retrieve
Date	: 23/05/2016
Time	: 16:03:59
Error	: Metadata contains a reference that cannot be resolved: ../XRMServices/2011/Discovery.svc?wsdl&sdkversion=0'.
Stack Trace	: at System.ServiceModel.Description.MetadataExchangeClient.MetadataRetriever.Retrieve(TimeoutHelper timeoutHelper)
   at System.ServiceModel.Description.MetadataExchangeClient.ResolveNext(ResolveCallState resolveCallState)
   at System.ServiceModel.Description.MetadataExchangeClient.GetMetadata(MetadataRetriever retriever)
   at Microsoft.Xrm.Sdk.Client.ServiceMetadataUtility.RetrieveServiceEndpointMetadata(Type contractType, Uri serviceUri, Boolean checkForSecondary)
   at Microsoft.Xrm.Sdk.Client.ServiceConfiguration`1..ctor(Uri serviceUri, Boolean checkForSecondary)
   at Microsoft.Xrm.Sdk.Client.ServiceConfigurationFactory.CreateManagement[TService](Uri serviceUri, Boolean enableProxyTypes, Assembly assembly)
   at Microsoft.Xrm.Tooling.Connector.CrmWebSvc.CreateAndAuthenticateProxy[T](IServiceManagement`1 servicecfg, Uri ServiceUri, Uri homeRealm, ClientCredentials userCredentials, ClientCredentials deviceCredentials, String LogString)
   at Microsoft.Xrm.Tooling.Connector.CrmWebSvc.DiscoverOrganizations(Uri discoveryServiceUri, Uri homeRealmUri, ClientCredentials clientCredentials, ClientCredentials deviceCredentials)
   at Microsoft.Xrm.Tooling.Connector.CrmWebSvc.InitCRM2011Service()
======================================================================================================================
Inner Exception Level 1	: 
Source	: System
Method	: GetResponse
Date	: 23/05/2016
Time	: 16:03:59
Error	: The remote name could not be resolved: '..'
Stack Trace	: at System.Net.HttpWebRequest.GetResponse()
   at System.ServiceModel.Description.MetadataExchangeClient.MetadataLocationRetriever.DownloadMetadata(TimeoutHelper timeoutHelper)
   at System.ServiceModel.Description.MetadataExchangeClient.MetadataRetriever.Retrieve(TimeoutHelper timeoutHelper)
======================================================================================================================

Microsoft.Xrm.Tooling.Connector.CrmServiceClient Error: 2 : Unable to Login to Dynamics CRM
Unable to Login to Dynamics CRM
CrmSvcUtil Error: 2 : Exiting program with exit code 2 due to exception : System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Exception: Connection to CRM is not established. Aborting process.
   at Microsoft.Crm.Services.Utility.SdkMetadataProviderService.Microsoft.Crm.Services.Utility.IMetadataProviderService.LoadMetadata()
   at DLaB.CrmSvcUtilExtensions.Entity.MetadataProviderService..ctor(IMetadataProviderService defaultService, IDictionary`2 paramters)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Crm.Services.Utility.ServiceFactory.CreateInstance[TIService](TIService defaultServiceInstance, String parameterValue, CrmSvcUtilParameters parameters)
   at Microsoft.Crm.Services.Utility.ServiceProvider.InitializeServices(CrmSvcUtilParameters parameters)
   at Microsoft.Crm.Services.Utility.CrmSvcUtil.get_ServiceProvider()
   at Microsoft.Crm.Services.Utility.CrmSvcUtil.Run()
   at Microsoft.Crm.Services.Utility.CrmSvcUtil.Main(String[] args)
CrmSvcUtil Error: 2 : ===== DETAIL ======
CrmSvcUtil Error: 2 : Source	: mscorlib
Method	: InvokeMethod
Date	: 16:03:59
Time	: 23/05/2016
Error	: Exception has been thrown by the target of an invocation.
Stack Trace	: at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Crm.Services.Utility.ServiceFactory.CreateInstance[TIService](TIService defaultServiceInstance, String parameterValue, CrmSvcUtilParameters parameters)
   at Microsoft.Crm.Services.Utility.ServiceProvider.InitializeServices(CrmSvcUtilParameters parameters)
   at Microsoft.Crm.Services.Utility.CrmSvcUtil.get_ServiceProvider()
   at Microsoft.Crm.Services.Utility.CrmSvcUtil.Run()
   at Microsoft.Crm.Services.Utility.CrmSvcUtil.Main(String[] args)
======================================================================================================================
Inner Exception Level 1	: 
Source	: CrmSvcUtil
Method	: Microsoft.Crm.Services.Utility.IMetadataProviderService.LoadMetadata
Date	: 16:03:59
Time	: 23/05/2016
Error	: Connection to CRM is not established. Aborting process.
Stack Trace	: at Microsoft.Crm.Services.Utility.SdkMetadataProviderService.Microsoft.Crm.Services.Utility.IMetadataProviderService.LoadMetadata()
   at DLaB.CrmSvcUtilExtensions.Entity.MetadataProviderService..ctor(IMetadataProviderService defaultService, IDictionary`2 paramters)
======================================================================================================================

Output file was not updated or not found!  ..\XrmToolBox\Plugins\CrmSvcUtil Ref\CrmServiceContext.cs

**Comment by daryllabar** [comment]
Are the Domain, userName, and password all correct in the Commandline that gets generated?

**Comment by JurgenPostelmans** [comment]
I'm having the exact same issue.

**Comment by PeulenF** [comment]
I think the error message in my last comment was of several tries I made. I edited it to give a better view.
The connectionstring in the commandline seems to be incomplete, as the UserName doesn't get populated.

**Comment by daryllabar** [comment]
Can you try copying the command line generated in the log to an actual command lineand, add the missing username, and then try running it?  You can try the different URL names as well.  If you get something to work, let me know and I can edit the output to match.

**Comment by PeulenF** [comment]
It does work adding the username and changing the URL manually.

**Comment by daryllabar** [comment]
What is the issue with the URL?  Is it using the unique org Id?  Is the domain being populated correctly?

**Comment by JurgenPostelmans** [comment]
In my case it does not work. This is part of the output. This is coming from an isolated VM so I don't mind the password being in here :)

C:\Users\Administrator\Downloads\XrmToolBox\Plugins\CrmSvcUtil Ref>CrmSvcUtil.ex
e  /namespace:"CrmEarlyBound" /out:"C:\Users\Administrator\Downloads\XrmToolBox\
Plugins\CrmSvcUtil Ref\Entities.cs" /servicecontextname:"CrmServiceContext" /cod
ecustomization:"DLaB.CrmSvcUtilExtensions.Entity.CustomizeCodeDomService,DLaB.Cr
mSvcUtilExtensions" /codegenerationservice:"DLaB.CrmSvcUtilExtensions.Entity.Cus
tomCodeGenerationService,DLaB.CrmSvcUtilExtensions" /codewriterfilter:"DLaB.CrmS
vcUtilExtensions.Entity.CodeWriterFilterService,DLaB.CrmSvcUtilExtensions" /nami
ngservice:"DLaB.CrmSvcUtilExtensions.NamingService,DLaB.CrmSvcUtilExtensions" /m
etadataproviderservice:"DLaB.CrmSvcUtilExtensions.Entity.MetadataProviderService
,DLaB.CrmSvcUtilExtensions" /url:http://springfield:5555/U2U/XrmServices/2011/Or
ganization.svc /username:administrator /password:secret /domain:U2UCOURSE
CrmSvcUtil : CRM Service Utility [Version 8.0.1.7297]
© 2015 Microsoft Corporation. All rights reserved

Microsoft.Xrm.Tooling.Connector.CrmServiceClient Error: 2 : Organization not fou
nd. Org = U2U/
Microsoft.Xrm.Tooling.Connector.CrmServiceClient Error: 2 : Unable to Login to D
ynamics CRM
Unable to Login to Dynamics CRM

Exiting program with exception: Exception has been thrown by the target of an in
vocation.
CrmSvcUtil Error: 2 : Exiting program with exit code 2 due to exception : System
.Reflection.TargetInvocationException: Exception has been thrown by the target o
f an invocation. ---> System.Exception: Connection to CRM is not established. Ab
orting process.
   at Microsoft.Crm.Services.Utility.SdkMetadataProviderService.Microsoft.Crm.Se

**Comment by JurgenPostelmans** [comment]
And now the weir thing: if I copy all the files from the latest version of the CRM SDK into the C:\Users\Administrator\Downloads\XrmToolBox\Plugins\CrmSvcUtil Ref folder it will generate the entity classes

**Comment by PeulenF** [comment]
the organization URL for our test environment is http://crm.company.com/crmtest  and in the generated command line this was http://crmtest.company.com/crmtest

**Comment by daryllabar** [comment]
So @JurgenPostelmans, are you good?

**Comment by daryllabar** [comment]
@PeulenF ok, so it looks like I'm not generating that URL correctly..  I'll have to review that, as well as make sure I'm pulling the username.

**Comment by daryllabar** [comment]
Can you download, unblock, then unzip this folder into your XrmToolBox Plugin folder? and let me know if the URL is being created correctly?

[EarlyBoundGen.zip](https://github.com/daryllabar/DLaB.Xrm.XrmToolBoxTools/files/281984/DLaB.Xrm.2016.zip)

**Comment by PeulenF** [comment]
I'm unable to use the plugin using these files. Followed by a crash and the following error:

************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at DLaB.EarlyBoundGenerator.EarlyBoundGeneratorPlugin.HydrateSettingsFromUI()
   at DLaB.EarlyBoundGenerator.EarlyBoundGeneratorPlugin.ClosingPlugin(PluginCloseInfo info)
   at XrmToolBox.MainForm.RequestCloseTabs(IEnumerable`1 pages, PluginCloseInfo info)
   at XrmToolBox.MainForm.MainForm_FormClosing(Object sender, FormClosingEventArgs e)
   at System.Windows.Forms.Form.OnFormClosing(FormClosingEventArgs e)
   at System.Windows.Forms.Form.WmClose(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

**Comment by JurgenPostelmans** [comment]
I'm getting the same exception:

 the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at DLaB.EarlyBoundGenerator.EarlyBoundGeneratorPlugin.HydrateSettingsFromUI()
   at DLaB.EarlyBoundGenerator.EarlyBoundGeneratorPlugin.CreateCode(CreationType creationType)
   at DLaB.EarlyBoundGenerator.EarlyBoundGeneratorPlugin.Create(CreationType creationType)
   at DLaB.EarlyBoundGenerator.EarlyBoundGeneratorPlugin.BtnCreateEntities_Click(Object sender, EventArgs e)
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

**Comment by daryllabar** [comment]
I'll have to look at this deeper.  Thanks for trying...

**Comment by daryllabar** [comment]
@JurgenPostelmans can you get the latest version an try again?

**Comment by JurgenPostelmans** [comment]
@daryllabar I just tried the new version and it could successfully generate the entities, option sets and actions in C#. 

One strange thing however was that in my connection settings in XrmToolbox I have "Use your current credentials" checked and your plugin asked me for the password when I pressed the "Create All" button.
In the output below you can see that Username is empty in the connection string but password is there. Cloud be that this is the expected behavior?

CRM Authentication Type Detected: ActiveDirectory
Executing "C:\Users\Administrator\Downloads\XrmToolBox\Plugins\CrmSvcUtil Ref\crmsvcutil.exe" /generateActions /out:"C:\Users\Administrator\Downloads\XrmToolBox\Plugins\CrmSvcUtil Ref\Actions.cs" /namespace:"CrmEarlyBound" /codecustomization:"DLaB.CrmSvcUtilExtensions.Action.CustomizeCodeDomService,DLaB.CrmSvcUtilExtensions" /codegenerationservice:"DLaB.CrmSvcUtilExtensions.Action.CustomCodeGenerationService,DLaB.CrmSvcUtilExtensions" /codewriterfilter:"DLaB.CrmSvcUtilExtensions.Action.CodeWriterFilterService,DLaB.CrmSvcUtilExtensions" /connectionstring:"Url=http://springfield:5555/U2U;UserName=;Password=******" 
Executing "C:\Users\Administrator\Downloads\XrmToolBox\Plugins\CrmSvcUtil Ref\crmsvcutil.exe" /namespace:"CrmEarlyBound" /out:"C:\Users\Administrator\Downloads\XrmToolBox\Plugins\CrmSvcUtil Ref\OptionSets.cs" /codecustomization:"DLaB.CrmSvcUtilExtensions.OptionSet.CreateOptionSetEnums,DLaB.CrmSvcUtilExtensions" /codegenerationservice:"DLaB.CrmSvcUtilExtensions.OptionSet.CustomCodeGenerationService,DLaB.CrmSvcUtilExtensions" /codewriterfilter:"DLaB.CrmSvcUtilExtensions.OptionSet.CodeWriterFilterService,DLaB.CrmSvcUtilExtensions" /namingservice:"DLaB.CrmSvcUtilExtensions.NamingService,DLaB.CrmSvcUtilExtensions" /connectionstring:"Url=http://springfield:5555/U2U;UserName=;Password=******"

**Comment by daryllabar** [comment]
Closing this in favor of new #47 since multiple issues were discussed in this issue.
